### ISSUE TITLE
Mistaken "Declaration of public static member variable" error

### ISSUE THREAD

**Issue by levibotelho** [palantir/tslint#1250]
### Bug Report

- __TSLint version__: `3.10.1`
- __TypeScript version__: `1.8`
- __Running TSLint via__: gulp-tslint & VS Code tslint extension

#### TypeScript code being linted

```ts
    static contextTypes = new class extends ValidationMap {
        router = React.PropTypes.object.isRequired;
    };

    static propTypes = new class extends ValidationMap {
        message = React.PropTypes.object.isRequired;
    };
```

with `tslint.json`:

#### Actual behavior

I get a "Declaration of public static member variable not allowed to appear after declaration of public instance member variable" error on the second instance. This also occurs if I switch the two declarations around.

#### Expected behavior

There should be no error.

**Comment by adidahiya** [comment]
Looks like we probably don't handle class expressions properly. The fix is to add this method to `MemberOrderingWalker`:

```ts
    public visitClassExpression(node: ts.ClassExpression) {
        this.resetPreviousModifiers();

        this.newMemberList();
        super.visitClassExpression(node);
        this.checkMemberOrder();
    }
```

and make sure the appropriate `SyntaxWalker` super method is defined, etc.
